File: examples/python/optimize.py
"""Plan routes for a fleet of delivery vehicles."""

from osrmplus import Client

client = Client("your_api_key")

plan = client.optimize(
    vehicles=[
        {
            "id": 1,
            "start": [74.3436, 31.5497],
            "end": [74.3436, 31.5497],
            "capacity": [20],
        },
        {
            "id": 2,
            "start": [74.3436, 31.5497],
            "end": [74.3436, 31.5497],
            "capacity": [20],
        },
    ],
    jobs=[
        {"id": 1, "location": [74.2650, 31.4700], "delivery": [5], "service": 300},
        {"id": 2, "location": [74.3000, 31.5000], "delivery": [8], "service": 300},
        {"id": 3, "location": [74.2800, 31.5200], "delivery": [3], "service": 300},
        {"id": 4, "location": [74.3200, 31.5100], "delivery": [6], "service": 300},
        {"id": 5, "location": [74.2900, 31.4800], "delivery": [4], "service": 300},
    ],
)

for r in plan["routes"]:
    stops = [s for s in r["steps"] if s["type"] == "job"]
    print(f"Vehicle {r['vehicle']}: {len(stops)} deliveries, cost {r['cost']}")
    for s in stops:
        print(f"  -> Job {s['id']} at {s['location']}")

if plan["unassigned"]:
    print(f"\n{len(plan['unassigned'])} jobs could not be assigned")

File: examples/javascript/optimize.js
import { OSRMPlus } from "osrmplus";

const client = new OSRMPlus("your_api_key");

// Coordinates are [lat, lon]. Index 0 is the depot.
const result = await client.vrp({
  coordinates: [
    [31.5497, 74.3436],   // Depot (Gulberg)
    [31.47, 74.265],      // Stop 1 (Model Town)
    [31.5, 74.3],         // Stop 2 (Liberty)
    [31.52, 74.28],       // Stop 3 (Johar Town)
    [31.51, 74.32],       // Stop 4
    [31.48, 74.29],       // Stop 5
  ],
  num_vehicles: 2,
  depot: 0,
  demands: [0, 3, 2, 4, 6, 4],
  vehicle_capacities: [10, 10],
  time_limit_seconds: 10,
});

console.log(`Status: ${result.status}`);
console.log(`Total distance: ${result.total_distance}`);
console.log(`Vehicles used: ${result.statistics.vehicles_used}`);

for (const r of result.routes) {
  console.log(`  Vehicle ${r.vehicle_id}: [${r.route}] (distance ${r.distance})`);
}

if (result.dropped_nodes.length) {
  console.log(`Dropped stops: ${result.dropped_nodes}`);
}
